contracts/RadiCards.sol

```diff
@@ -3,234 +3,504 @@ pragma solidity ^0.4.24;
-  }
+    event BenefactorAdded(
+        uint256 indexed _benefactorIndex
+    );
 
-  function setActive(uint256 _cardIndex, bool _active) external onlyIfWhitelisted(msg.sender) {
-    require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+    event CardAdded(
+        uint256 indexed _cardIndex
+    );
 
-    cards[_cardIndex].active = _active;
-  }
+    constructor () public ERC721Token("RadiCards", "RADI") {
+        addAddressToWhitelist(msg.sender);
+    }
+
+    function gift(address to, uint256 _benefactorIndex, uint256 _cardIndex, string _message, uint256 _donationAmount, uint256 _giftAmount, bool _claimableLink) payable public returns (bool) {
+        require(to != address(0), "Must be a valid address");
+        require(benefactors[_benefactorIndex].ethAddress != address(0), "Must specify existing benefactor");
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        require(cards[_cardIndex].active, "Must be an active card");
+
+        Statuses _giftStatus;
+        address _sentToAddress;
+
+        if(_claimableLink){
+            require(_donationAmount + _giftAmount + EPHEMERAL_ADDRESS_FEE == msg.value, "Can only request to donate and gift the amount of ether sent + Ephemeral fee");
+            _giftStatus = Statuses.Deposited;
+            _sentToAddress = this;
+            ephemeralWalletCards[to] = tokenIdPointer;
+            to.transfer(EPHEMERAL_ADDRESS_FEE);
+        }
+
+        else {
+            require(_donationAmount + _giftAmount == msg.value,"Can only request to donate and gift the amount of ether sent");
+            _giftStatus = Statuses.Claimed;
+            _sentToAddress = to;
+        }
+
+        if (cards[_cardIndex].maxQnty > 0){ //the max quantity is set to zero to indicate no limit. Only need to check that can mint if limited
+            require(cards[_cardIndex].minted < cards[_cardIndex].maxQnty, "Can't exceed maximum quantity of card type");
+        }
+
+        if(cards[_cardIndex].minPrice > 0){ //if the card has a minimum price check that enough has been sent
+        // Convert the current value of the eth send to a USD value of atto (1 usd = 10^18 atto).
+        // require(getEthUsdValue(msg.value) >= (cards[_cardIndex].minPrice), "Must send at least the minimum amount");
+            require (getMinCardPriceInWei(_cardIndex)<=msg.value,"Must send at least the minimum amount to buy card");
+        }
+
+        tokenIdToRadiCardIndex[tokenIdPointer] = RadiCard({
+            gifter : msg.sender,
+            message : _message,
+            daiDonation: false,
+            giftAmount : _giftAmount,
+            donationAmount: _donationAmount,
+            status: _giftStatus,
+            cardIndex : _cardIndex,
+            benefactorIndex : _benefactorIndex
+        });
+
+        // Card is minted to the _sentToAddress. This is either this radicards contract(if claimableLink==true)
```

Comment:
the comments here are not clarifying it for me. Maybe just say "this could be a ephermeral address"